pkg/plugins/manager/loader/initializer/initializer.go

```diff
@@ -106,5 +126,23 @@ func getPluginSettings(pluginID string, cfg *config.Cfg) pluginSettings {
 		ps[k] = v
 	}
 
-	return ps
+	return ps, nil
+}
+
+func (i *Initializer) oauth2OnBehalfOfVars(pluginID string, oauthAppInfo *oauthserver.ExternalServiceRegistration) ([]string, error) {
+	cli, err := i.oauthServer.SaveExternalService(context.Background(), &oauthserver.ExternalServiceRegistration{
+		ExternalServiceName:    pluginID,
+		Permissions:            oauthAppInfo.Permissions,
+		ImpersonatePermissions: oauthAppInfo.ImpersonatePermissions,
+		Key:                    oauthAppInfo.Key,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return []string{
+		fmt.Sprintf("GF_PLUGIN_APP_CLIENT_ID=%s", cli.ID),
+		fmt.Sprintf("GF_PLUGIN_APP_CLIENT_SECRET=%s", cli.Secret),
+		fmt.Sprintf("GF_PLUGIN_APP_PRIVATE_KEY=%s", cli.KeyResult.PrivatePem),
```

Comment:
Would it make sense to pass the generated private key's type (RSA/ECDSA) to the plugin in case it was generated by Grafana? (`s.cfg.OAuth2ServerGeneratedKeyTypeForClient`)

Reply:
I am not sure about this. If I simply forward that value, it would be a bit arbitrary. Since it's not an input nor an output for `SaveExternalService` it would be difficult to know why that is relevant. I would either:

- Make it part of the response. Something like `cli.Key.Algorithm`.
- Let the client (plugin / sdk) solve it. Right now, it's already differing between a RSA or ECDSA key based on the private key (`RSA PRIVATE KEY` vs `PRIVATE KEY` ([link](https://github.com/grafana/grafana-plugin-examples/pull/138/files#diff-3592b85781284388d41ec7641f7110d4ab2de2dd52506bc283781c7b7bc5d83aR84-R99))).

WDYT?

Reply:
Yep, that's why I asked because I was not sure, but if this check will be hidden from the plugin developer in the sdk (2. option) then that would be the best!